internal/controllers/cryostat_controller.go

```diff
@@ -710,14 +710,18 @@ func (r *CryostatReconciler) createOrUpdateDeployment(ctx context.Context, deplo
 		if err := controllerutil.SetControllerReference(owner, deploy, r.Scheme); err != nil {
 			return err
 		}
-		// Update pod template spec and selector to propagate any changes from Cryostat CR
-		deploy.Spec.Template.Spec = specCopy.Template.Spec
-		// FIXME Immutable
-		deploy.Spec.Selector = specCopy.Selector
+		// Immutable, only updated when the deployment is created
+		if deploy.ObjectMeta.CreationTimestamp.IsZero() {
```

Comment:
I think you can do `deploy.CreationTimestamp.IsZero()` to make this shorter. I've also filed #433, which relates to this problem. This is part of the fix, but we should also detect when the selector needs to be updated and delete/recreate the deployment as a workaround.